### ISSUE TITLE
Option to not pause video when playing thumbnail preview

### ISSUE THREAD

**Issue by alpe12** [rotemdan/MaximizerForYouTube#10]
I like that it pauses the playing video when another video starts playing.
But it shouldn't stop playing the video when I mouse-over a video and the "thumbnail" animates (showing video preview).
I would like an option to make it not pause on this specific interaction only (thumbnail).
Thanks anyway.

**Comment by rotemdan** [comment]
Thanks for the report, but I believe it already was already fixed it in [this August 19 2023 commit](https://github.com/rotemdan/MaximizerForYouTube/commit/5f420f9fa0d210558f991da83e69038f0fd98954) titled "Don't auto-pause playing video if currently on home page, feed page or a search page". Which also changed the version to `0.2.4`.

The changed code looked like:
```js
	if (videoPlayer.length > 0 &&
		!videoPlayer.hasClass("MaximizerForYouTube_Modded_Autopause") &&
		!inHomePage() &&
		!inFeedPage() &&
		!inSearchPage()) {
```

I don't see this happening anymore (and I do use the extension daily). Maybe there are other YouTube pages that it happens in, or you are using an older version?

**Comment by alpe12** [comment]
Maybe it could use:
`new URL(location.href).pathname === '/'`

It would math both with and without / at the end. And will also ignore any query string that the link might have.

**Comment by rotemdan** [comment]
Thanks for the suggestion. I changed to:

```js
function inWatchPage() {
	return locationPathname() == '/watch'
}

function inSearchPage() {
	return locationPathname().startsWith('/results')
}

function inFeedPage() {
	return locationPathname().startsWith('/feed')
}

function inHomePage() {
	return locationPathname() == '/'
}

function locationPathname() {
	return (new URL(location.href)).pathname
}
```
It seems to work so far, but I can't see or verify the homepage since I disabled YouTube history (or either I block it with an extension, not sure). Let me know if there are other special pages that should be checked for.

Anyway, this script was mostly written in 2017 (there was an earlier version from several years earlier). It got almost no updates since 2019.

I published `0.2.7` with the changes.

**Comment by alpe12** [comment]
Thanks.
